### DISCUSSION TITLE
Last encoding failure

### DISCUSSION THREAD

**Discussion by brycefernandes** [nats-io/nats.deno#728] [Q&A]
Hi , I'm stuck at a place. I've tried looking for examples specific to js but no luck
Here is the code. 
`const nats = require("nats");
const { encode, decode } = require("nats-jwt");

class AuthService {
  constructor(issuerKeyPair, handler) {
    this.issuerKeyPair = issuerKeyPair;
    this.handler = handler;
  }

  async handle(request) {
    try {
      console.log("AuthService: Handling request", request.data.toString());
      const rc = decode(request.data.toString());
      if (!rc) {
        throw new Error("Invalid authorization request");
      }
      console.log("AuthService: Decoded request:", JSON.stringify(rc, null, 2));

      const claims = {
        sub: rc.sub,
        aud: "CHAT",
      };

      console.log("AuthService: Calling handler...");
      const handledClaims = await this.handler(rc);
      Object.assign(claims, handledClaims);

      console.log("AuthService: Validating and signing claims...");
      const token = await this.validateAndSign(claims);
      console.log("AuthService: Responding to request...");
      this.respond(request, rc.sub, rc.nats.server_id.id, token);
    } catch (err) {
      console.error("AuthService: Error handling auth request", err);
      request.respond(nats.Empty, { status: 500 });
    }
  }

  respond(request, userNKey, serverId, userJWT) {
    console.log("AuthService: Constructing response...");
    const rc = {
      sub: userNKey,
      aud: serverId,
      jwt: userJWT,
    };

    console.log("AuthService: Signing response...");
    const token = encode(rc, this.issuerKeyPair);
    console.log("AuthService: Sending response...");
    request.respond(Buffer.from(token));
  }

  async validateAndSign(claims) {
    console.log(
      "AuthService: Validating and signing claims:",
      JSON.stringify(claims, null, 2),
    );
    return encode(claims, this.issuerKeyPair);
  }
}

module.exports = AuthService;
`

I get an error on this line
`    return encode(claims, this.issuerKeyPair);
`

What are we supposed to do here?  I'm sure there is a method, but I can figure it out.
Regards

**Comment by aricart** [comment]
Are you trying to implement a call out?

**Reply by aricart** [reply]
If you are trying to implement a callout, perhaps take a look at this (in particular the tests).
https://github.com/synadia-io/orbit.js/tree/authcallout/callout

Note that the above branch is a WIP, so things will change.

**Comment by aricart** [comment]
As for your description above, it would be very helpful to explain what you are trying to do and the exact error you are getting.
